`gsKit` is not compiled in Ubuntu

I have downloaded the latest artefacts and I see how the `gsKit` folder is not in the `Ubuntu` toolchains (neither x86 and/or arm one)

We can observe the `macos` one:
```
➜  ps2dev_macos ls -ll
total 0
drwxr-xr-x@  6 fjtrujy  staff  192 10 dic.  04:22 bin
drwxr-xr-x@  5 fjtrujy  staff  160 10 dic.  03:20 dvp
drwxr-xr-x@  8 fjtrujy  staff  256 10 dic.  03:56 ee
drwxr-xr-x@  4 fjtrujy  staff  128 10 dic.  04:18 gsKit
drwxr-xr-x@  8 fjtrujy  staff  256 10 dic.  03:39 iop
drwxr-xr-x@ 18 fjtrujy  staff  576 10 dic.  04:12 ps2sdk
drwxr-xr-x@  5 fjtrujy  staff  160 10 dic.  04:22 share
-rw-r--r--@  1 fjtrujy  staff    0 10 dic.  03:18 test.tmp
```

Then on `ubuntu-arm`:
```bash
➜  ps2dev_ubuntu_arm ls -ll
total 0
drwxr-xr-x@  5 fjtrujy  staff  160 10 dic.  03:19 dvp
drwxr-xr-x@  8 fjtrujy  staff  256 10 dic.  03:44 ee
drwxr-xr-x@  8 fjtrujy  staff  256 10 dic.  03:32 iop
drwxr-xr-x@ 19 fjtrujy  staff  608 10 dic.  10:42 ps2sdk
drwxr-xr-x@  4 fjtrujy  staff  128 10 dic.  03:55 share
-rw-r--r--@  1 fjtrujy  staff    0 10 dic.  03:19 test.tmp
```

And also for `ubuntu_x64`
```
➜  ps2dev_ubuntu_x64 ls -ll
total 0
drwxr-xr-x@  5 fjtrujy  staff  160 10 dic.  03:19 dvp
drwxr-xr-x@  8 fjtrujy  staff  256 10 dic.  03:54 ee
drwxr-xr-x@  8 fjtrujy  staff  256 10 dic.  03:37 iop
drwxr-xr-x@ 18 fjtrujy  staff  576 10 dic.  04:10 ps2sdk
drwxr-xr-x@  4 fjtrujy  staff  128 10 dic.  04:10 share
-rw-r--r--@  1 fjtrujy  staff    0 10 dic.  03:18 test.tmp
```


In the Windows artefacts it is being compiled and it appears in the list of folders.

I suppose it is working as expected in the Alpine machine, otherwise all the workflow for other apps would be failing

I have been watching the generated logs for ubuntu and I see:

```
2025-12-10T03:10:24.2073858Z make[2]: Leaving directory '/__w/ps2dev/ps2dev/build/ps2sdk-ports/build/zlib/build_ee'
2025-12-10T03:10:24.2326047Z Install the project...
2025-12-10T03:10:24.2446231Z -- Install configuration: "RelWithDebInfo"
2025-12-10T03:10:24.2473166Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib/libz.a
2025-12-10T03:10:24.2474095Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib/cmake/zlib/ZLIB-static.cmake
2025-12-10T03:10:24.2475272Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib/cmake/zlib/ZLIB-static-relwithdebinfo.cmake
2025-12-10T03:10:24.2476476Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib/cmake/zlib/ZLIBConfig.cmake
2025-12-10T03:10:24.2477532Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib/cmake/zlib/ZLIBConfigVersion.cmake
2025-12-10T03:10:24.2480969Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/include/zconf.h
2025-12-10T03:10:24.2485943Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/include/zlib.h
2025-12-10T03:10:24.2493275Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/man/man3/zlib.3
2025-12-10T03:10:24.2499378Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/doc/zlib/zlib/LICENSE
2025-12-10T03:10:24.2523705Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/doc/zlib/zlib/algorithm.txt
2025-12-10T03:10:24.2524828Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/doc/zlib/zlib/crc-doc.1.0.pdf
2025-12-10T03:10:24.2525860Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/doc/zlib/zlib/rfc1950.txt
2025-12-10T03:10:24.2526923Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/doc/zlib/zlib/rfc1951.txt
2025-12-10T03:10:24.2532173Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/doc/zlib/zlib/rfc1952.txt
2025-12-10T03:10:24.2536528Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/share/doc/zlib/zlib/txtvsbin.txt
2025-12-10T03:10:24.2542105Z -- Installing: /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib/pkgconfig/zlib.pc
2025-12-10T03:10:24.2564759Z make[1]: Leaving directory '/__w/ps2dev/ps2dev/build/ps2sdk-ports/build/zlib/build_ee'
2025-12-10T03:10:24.2586557Z Building 'xz' for EE...
2025-12-10T03:10:24.2720391Z CMake Error at CMakeLists.txt:80 (cmake_minimum_required):
2025-12-10T03:10:24.2721232Z   CMake 3.20 or higher is required.  You are running version 3.16.3
2025-12-10T03:10:24.2721763Z 
2025-12-10T03:10:24.2722085Z 
2025-12-10T03:10:24.2726469Z -- Configuring incomplete, errors occurred!
2025-12-10T03:10:24.2748637Z make: *** [Makefile:55: cmakelibs] Error 1
2025-12-10T03:10:24.4405069Z mips64r5900el-ps2-elf-ar rc liblua.a lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o ltests.o lauxlib.o lbaselib.o ldblib.o liolib.o lmathlib.o loslib.o ltablib.o lstrlib.o lutf8lib.o loadlib.o lcorolib.o linit.o
2025-12-10T03:10:24.4485870Z mips64r5900el-ps2-elf-ranlib liblua.a
2025-12-10T03:10:24.4561684Z mips64r5900el-ps2-elf-gcc -o lua  lua.o liblua.a -lm  
2025-12-10T03:10:24.7209069Z touch all
2025-12-10T03:10:24.7215935Z make[1]: Leaving directory '/__w/ps2dev/ps2dev/build/ps2sdk-ports/build/lua'
2025-12-10T03:10:24.7218734Z make -C build/lua install platform=PS2
2025-12-10T03:10:24.7245540Z make[1]: Entering directory '/__w/ps2dev/ps2dev/build/ps2sdk-ports/build/lua'
2025-12-10T03:10:24.7246164Z mkdir -p /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/include
2025-12-10T03:10:24.7255178Z mkdir -p /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib
2025-12-10T03:10:24.7264943Z cp lua.h luaconf.h lualib.h lauxlib.h /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/include
2025-12-10T03:10:24.7279037Z cp lua liblua.a /__w/ps2dev/ps2dev/ps2dev/ps2sdk/ports/lib
2025-12-10T03:10:24.7342114Z make[1]: Leaving directory '/__w/ps2dev/ps2dev/build/ps2sdk-ports/build/lua'
2025-12-10T03:10:24.7348333Z 
2025-12-10T03:10:24.7348369Z 
2025-12-10T03:10:24.7348374Z 
2025-12-10T03:10:24.7348379Z 
2025-12-10T03:10:24.7348513Z 
2025-12-10T03:10:24.7348521Z 
2025-12-10T03:10:24.7348543Z 
2025-12-10T03:10:24.7348548Z 
2025-12-10T03:10:24.7348553Z 
2025-12-10T03:10:24.7348558Z 
2025-12-10T03:10:24.7348562Z 
2025-12-10T03:10:24.7348566Z 
2025-12-10T03:10:24.7348571Z 
2025-12-10T03:10:24.7348576Z 
2025-12-10T03:10:24.7348646Z 
2025-12-10T03:10:24.7348651Z 
2025-12-10T03:10:24.7348667Z 
2025-12-10T03:10:24.7348672Z 
2025-12-10T03:10:24.7348677Z 
2025-12-10T03:10:24.7348692Z 
2025-12-10T03:10:24.7348697Z 
2025-12-10T03:10:24.7348701Z 
2025-12-10T03:10:24.7348714Z 
2025-12-10T03:10:24.7348729Z 
2025-12-10T03:10:24.7348920Z ../scripts/003-ps2sdk-ports.sh: Failed.
2025-12-10T03:10:24.7349327Z ===========================
2025-12-10T03:10:24.7349925Z Please see the community links at https://ps2dev.github.io/#community for assistance
2025-12-10T03:10:24.7350514Z ===========================
```

It is also failing for `libconfuse`:
```
-fno-stack-protector -fno-common -march=r5900 -I/__w/ps2dev/ps2dev/ps2dev/ps2sdk/ee/include -I/__w/ps2dev/ps2dev/ps2dev/ps2sdk/common/include -I.  -c ldump.c -o ldump.o
2025-12-10T03:10:19.4705309Z configure.ac: installing 'support/ylwrap'
2025-12-10T03:10:19.5299110Z parallel-tests: installing 'support/test-driver'
2025-12-10T03:10:19.5355671Z -- Looking for fseeko - found
2025-12-10T03:10:19.5358524Z -- Looking for stdarg.h
2025-12-10T03:10:19.5742780Z autoreconf: automake failed with exit status: 1
2025-12-10T03:10:19.5772217Z make: *** [Makefile:62: libconfuse] Error 1
2025-12-10T03:10:19.5788413Z make: *** Waiting for unfinished jobs....
```

Why are we ignoring the compilation errors on the `ps2sdk-ports` stage?

@uyjulian, please could you take a look at this?

--- a/.github/workflows/compilation.yml
+++ b/.github/workflows/compilation.yml
@@ -81,7 +81,11 @@ jobs:
       if: startsWith(matrix.os.runs-on, 'ubuntu')
       run: |
         apt-get -y update
-        apt-get -y -o Dpkg::Options::='--force-confdef' -o Dpkg::Options::='--force-confold' install texinfo bison flex gettext autopoint autoconf automake libgmp3-dev libmpfr-dev libmpc-dev cmake g++ gcc git libgsl-dev make patch zlib1g-dev wget curl
+        apt-get -y -o Dpkg::Options::='--force-confdef' -o Dpkg::Options::='--force-confold' install texinfo bison flex gettext autopoint autoconf automake libtool libgmp3-dev libmpfr-dev libmpc-dev g++ gcc git libgsl-dev make patch zlib1g-dev wget curl ca-certificates gpg
+        # Install newer cmake (3.20+) from Kitware's official repository
+        # Ubuntu 20.04's default cmake is 3.16.3, but xz v5.8.1 requires 3.20+
+        curl -LsSf https://apt.kitware.com/kitware-archive.sh | sh
+        apt-get -y install cmake
 
     - name: Install macOS packages (brew)
       if: startsWith(matrix.os.runs-on, 'macos') && matrix.os.macos-package-manager == 'brew'
@@ -110,6 +114,7 @@ jobs:
 
     - name: Runs all the stages in the shell
       run: |
+        set -e
         . ./config/ci-env.sh
         if test -f ./build-all.sh; then ./build-all.sh; fi
         if test -f ./toolchain.sh; then ./toolchain.sh; fi

